app.rb

```diff
@@ -0,0 +1,95 @@
+require './person'
+require './teacher'
+require './student'
+require './book'
+require './rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'All Books:'
+    books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'All People:'
+    people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'To create a student type (1) and type (2) for a teacher? [Input the number]: '
+    select_person = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    case select_person
+    when 1
+      print 'Has parent permission? [Y/N]: '
+      @people << Student.new(age, nil, name: name, parent_permission: true)
+      puts 'Student created successfully'
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, specialization, name: name)
+      puts 'Teacher created successfully'
+    else
+      puts 'Invalid choice. Choose 1 for student or 2 for teacher.'
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
```

Comment:
- [ ] Kindly make sure the newly created book has a unique `id` which will help the user to be able to create the new `rental` successfully. At this moment the user cannot be able to create a `rental` because it requires inserting the book `id` which is not generated while creating a book. Kindly fix that.

  ### ERROR WHICH APPEARS EVERYTIME I TRY TO  CREATE A NEW RENTAL
  
     ![image](https://user-images.githubusercontent.com/87197412/232131212-04fd0dca-b185-47ab-a7f0-29c6fc159d74.png)

  ### LIST OF BOOKS THAT CONFIRMS THAT NO ID WAS GENERATED BEFORE
  
    ![image](https://user-images.githubusercontent.com/87197412/232131726-9fac013b-eb93-4a9d-b2ad-3af8aecfd961.png)

Reply:
Fixed